app/models/health/cha_saver.rb

```diff
@@ -37,31 +33,30 @@ def create
       end
     end
 
-
     def update
       @cha.class.transaction do
         @cha.completed_at = nil unless @complete
         @cha.save!
-        if @complete && @reviewed && @create_qa
-          @qualifying_activity.source_id = @cha.id
-          @qualifying_activity.save
+        if @complete && @reviewed && @create_qa && @cha.patient.recent_ssm_form.completed_at.present?
```

Comment:
I'm confused, why is the CHA checking for a recent completed SSM?  Maybe a comment as to the intention is necessary.

Reply:
This QA requires both the CHA and the SSM to have been completed, and can be done in either order, so both the ChaSaver and the SsmSaver need to generate the QA, just whichever is last.